### PR TITLE
Code Updates / Refactors

### PR DESCRIPTION
**Pull request by TomDonoghue**

### CODE REVIEW

File: local/notebooks/analysis.py [hunk 1]
```diff
@@ -11,21 +11,35 @@
...
     """Runs an analysis across state for multiple dictionaries
     
-    datasets: list of dict of 4d arrays
+    datasets: list of dict
+        each list entry is a state
+            0 - Trial
+            1 - Rest
+        each dict entry is a band with 3d array of subject channel feature 
     label: str
     eeg_dat_info: str
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py]: These arguments aren't actually inputs to the function - can you drop them from the docs?

File: local/notebooks/analysis.py [hunk 2]
```diff
@@ -34,53 +48,61 @@ def run_state_dict(datasets, label, mask, save_fig):
 def run_state_array(datasets, label, mask, feats, save_fig=True):
     """Runs an analysis across state for multiple arrays
     
-    datasets: list of dict of 4d arrays
+    datasets: list of 3d arrays
     label: str
     eeg_dat_info: str
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py]: Same as above: get rid of extra arguments listed in docs.

File: local/notebooks/analysis.py [hunk 3]
```diff
@@ -11,21 +11,35 @@
 def run_state_dict(datasets, label, mask, save_fig):
     """Runs an analysis across state for multiple dictionaries
     
-    datasets: list of dict of 4d arrays
+    datasets: list of dict
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py:18]: I would write this as:
`list of dict, as [trial_data_dict, rest_data_dict]`
`each dict entry is a band with 3d array, as [subjects, channels, features]`

File: local/notebooks/analysis.py [hunk 4]
```diff
@@ -11,21 +11,35 @@
...
+    # extracting the band names from one of the lists
+    # "theta, alpha, beta"
+
+    feats = ["CFS", "PWS", "BWS"]
+    # creating a list of the feature names 
 
     corr_dicts = []
 
     for band in bands:
+    # This loop goes through each of the bands individual bands
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py:38]: In general, I would tend to (I think this is the convention) put docs above the line(s) they relate to.

File: local/notebooks/analysis.py [hunk 5]
```diff
@@ -11,21 +11,35 @@
...
+    # Resulting in a [n_subjects, num_blocks, n_channels, n_feats] 
 
         curr_data = [dataset[band] for dataset in datasets]
+        # curr_data is a list of [n_subjects, num_blocks, n_channels, n_feats] for both states
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py:42]: For consistency, when indicating the # of something, always `n_thing`.
So, `num_blocks` should be `n_blocks` (here and elsewhere).

File: local/notebooks/analysis.py [hunk 6]
```diff
@@ -11,21 +11,35 @@
 def run_state_dict(datasets, label, mask, save_fig):
     """Runs an analysis across state for multiple dictionaries
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py]: 'an analysis' (here and in other functions) isn't very specific. What is the analysis?

File: local/notebooks/analysis.py (hunk 2 above)
Comment (TomDonoghue) [reply local/notebooks/analysis.py]: Double check this - seems like datasets is a list of dict of 4d arrays? Check and make sure docs is right.

File: local/notebooks/analysis.py [hunk 7]
```diff
@@ -34,53 +48,61 @@ def run_state_dict(datasets, label, mask, save_fig):
...
             out_data = dataset[:, :, feat_in]
             outputs.append(out_data)
+            # Extract desired feature
+            # Resulting in output being 2d array [n_subjects, n_channels] 
 
         name = label + "_" + feat
-        state_ttest_dict[name] = ttest_ind(outputs[0], outputs[1])
-
+   
+        # QUESTION: Should I average ACROSS SUBJECTS, ACROSS CHANNELS??
```
Comment (TomDonoghue) [inline local/notebooks/analysis.py]: Average across channels (not across subjects - we compare across subjects).

**Comment by TomDonoghue** [comment]
Also to do on the PR, finish the things listed in #6

**Comment by LuyandaMdanda** [comment]
completed all work items. Need you to locally run FOOOF_plots.ipynb

**Pull request merged**